### PR TITLE
fix(next-mdx): resolve webpack loader

### PR DESCRIPTION
**Pull request by merceyz**
**What's the problem this PR addresses?**

`@next/mdx` adds the webpack loader `@mdx-js/loader` without resolving it to an absolute path

Depends on https://github.com/vercel/next.js/pull/17606

**How did you fix it?**

`require.resolve` the webpack loader before adding it

### CODE REVIEW

File: examples/with-mdx/package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
   },
   "dependencies": {
     "@mdx-js/loader": "^1.5.1",
+    "@mdx-js/react": "^1.6.18",
```
Comment (merceyz) [inline examples/with-mdx/package.json:11]: Required until https://github.com/mdx-js/mdx/pull/1300 is merged

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| buildDuration | 12.4s | 12.4s | ⚠️ +44ms |
| nodeModulesSize | 64.4 MB | 64.4 MB | ⚠️ +246 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.325 | 2.293 | -0.03 |
| / avg req/sec | 1075.16 | 1090.18 | +15.02 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.336 | 1.234 | -0.1 |
| /error-in-render avg req/sec | 1871.44 | 2026.39 | +154.95 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| 677f882d2ed8..7765.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-5aa1543..4e79.js gzip | 7.34 kB | 7.34 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.1 kB | 58.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.94 kB | 6.94 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-7d849dd..dule.js gzip | 6.32 kB | 6.32 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53 kB | 53 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-89ad9e7..25bb.js gzip | 1.34 kB | 1.34 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.74 kB | 7.74 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-aeb707b..dule.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.39 kB | 5.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 996 B | 996 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| buildDuration | 14.4s | 14.2s | -176ms |
| nodeModulesSize | 64.4 MB | 64.4 MB | ⚠️ +246 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| 677f882d2ed8..7765.js gzip | 11.1 kB | 11.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-5aa1543..4e79.js gzip | 7.34 kB | 7.34 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.1 kB | 58.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.94 kB | 6.94 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-7d849dd..dule.js gzip | 6.32 kB | 6.32 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53 kB | 53 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-89ad9e7..25bb.js gzip | 1.34 kB | 1.34 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.74 kB | 7.74 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-aeb707b..dule.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.39 kB | 5.39 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _error.js | 1.06 MB | 1.06 MB | ✓ |
| 404.html | 4.34 kB | 4.34 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 1.06 MB | 1.06 MB | ✓ |
| link.js | 1.1 MB | 1.1 MB | ✓ |
| routerDirect.js | 1.1 MB | 1.1 MB | ✓ |
| withRouter.js | 1.1 MB | 1.1 MB | ✓ |
| Overall change | 5.42 MB | 5.42 MB | ✓ |

</details>

</details>
Commit: 6817507fd15173ebbd9b735e163e5be5c4ff4b43

**Comment by Timer** [comment]
Should this PR be marked ready?

File: examples/with-mdx/package.json (hunk 1 above)
Comment (Timer) [reply examples/with-mdx/package.json:11]: Seems they said they wontfix.

Comment (merceyz) [reply examples/with-mdx/package.json:11]: Yeah, it's a shame they decided to go that route. Added it as a peer dependency to `@next/mdx` as that is essentially what it is.

**Comment by merceyz** [comment]
Yes, now that https://github.com/vercel/next.js/pull/17606 is merged this is ready

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| buildDuration | 9.1s | 9.3s | ⚠️ +185ms |
| nodeModulesSize | 80.6 MB | 80.6 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.784 | 1.763 | -0.02 |
| / avg req/sec | 1401.67 | 1417.69 | +16.02 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.121 | 1.136 | ⚠️ +0.01 |
| /error-in-render avg req/sec | 2230.37 | 2201.49 | ⚠️ -28.88 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| 677f882d2ed8..396f.js gzip | 13 kB | 13 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-fd59cc5..1b57.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.3 kB | 59.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-705099c..c35d.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| index.html gzip | 613 B | 613 B | ✓ |
| link.html gzip | 619 B | 619 B | ✓ |
| withRouter.html gzip | 607 B | 607 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| buildDuration | 10.8s | 10.9s | ⚠️ +171ms |
| nodeModulesSize | 80.6 MB | 80.6 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| 677f882d2ed8..396f.js gzip | 13 kB | 13 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-fd59cc5..1b57.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.3 kB | 59.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-705099c..c35d.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | merceyz/next.js merceyz/mdx | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.17 MB | 5.17 MB | ✓ |

</details>

</details>
Commit: ca4747e3c2c92564862d3a57973e3ff2c17bd31d

**Review by Timer — APPROVED** [review]

**Pull request merged**